hilti/toolchain/src/ast/ast-context.cc

```diff
@@ -570,6 +570,13 @@ static Result<Nothing> _runHook(bool* modified, const Plugin& plugin, PluginMemb
 }
 
 Result<Nothing> ASTContext::processAST(Builder* builder, Driver* driver) {
+    static auto report_ast_final = [&]() {
+        const auto& hilti_plugin = plugin::registry().hiltiPlugin();
+        _dumpAST(logging::debug::AstFinal, hilti_plugin, "Final AST", -1);
+        _dumpState(logging::debug::AstFinal);
+        _dumpStats(logging::debug::AstStats, hilti_plugin.component);
+    };
```

Comment:
It seems we could use `scope_exit` here.

```suggestion
    auto _guard = scope_exit([&]() {
        const auto& hilti_plugin = plugin::registry().hiltiPlugin();
        _dumpAST(logging::debug::AstFinal, hilti_plugin, "Final AST", -1);
        _dumpState(logging::debug::AstFinal);
        _dumpStats(logging::debug::AstStats, hilti_plugin.component);
    });
```

With that we should be able to drop the explicit calls below.

Reply:
Good idea!